File: Gaussian/Ju/Rando.jl
module Rando
export RandoData,fv
#=
Coefficients for a simple, but reproductible random number generator.
=#
mutable struct RandoData
    seed::Int64
    a::Int64
    c::Int64
    m::Int64
    RandoData()=new(123456789,1103515245,12345,4294967296)
end
#=
return a random float in [0,vmax]
=#
function fv!(R::RandoData,vmax=1.)
    R.seed= (R.a * R.seed + R.c) % R.m
    vmax*Float64(R.seed)/R.m
end
end

File: FeStiff/Pythran/rando.py
class Rando:
    def __init__(self):
        self.seed = 123456789
        self.a = 1103515245
        self.c = 12345
        self.m = 2 ** 32

    def get(self):
        self.seed = (self.a * self.seed + self.c) % self.m
        return self.seed

    def fv(self, vmax=1.0):
        return vmax * float(self.get()) / self.m


if __name__ == "__main__":
    R = rando()
    for i in range(100):
        print(R.fv(10.0))
